# How to perform PhageScope visualizations

PhageScope offers interactive visualizations for curated database and customized analysis results. The visualizations, which support informative tooltips to deliver detailed introductions and information, are prepared in an optional format with high quality, making them seamlessly incorporated into academic publications.  

We provide some demo visualizations in this tutorial.  

+ **The PhageScope visualization of phage completeness and phenotypes.** PhageScope provides visualizations for the overview of **a.** phage completeness, **b.** phage lifestyles, and **c.** host taxonomies.
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/OverviewDemo.png" width="600"></div>

+ **The PhageScope visualization of the ORFs and proteins on phage genomes.** PhageScope provides the linear (**a**) and circular (**b**) annotation visualizations. Diverse color codes signify the functional affiliation of each encoded protein; the arrow orientations demonstrate the transcriptional directionality; the blue polylines indicate the GC content within a sliding window of 30 bp; the red/green polylines indicate the GC skew within a sliding window of 30 bp.
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/ORFDemo.png" width="600"></div>

+ **The PhageScope visualization of the protein functional class distributions.** The upper heatmap showcases the percentage of ten functional classes for each phage, whereas the lower heatmap demonstrates the count of each functional class.
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/HeatmapDemo.png" width="600"></div>

+ **The PhageScope visualization of the transcriptional terminators.** Based on the structural annotation visualization, the outermost circle shows the locations of the predicted transcriptional terminators, with the color indicating the confidence level.
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/TermDemo.png" width="500"></div>

+ **The PhageScope visualization of the functional elements.** Based on the structural annotation visualization, the outermost circle shows the locations of the predicted tRNA and tmRNA (**a**), Anti-CRISPR proteins (**b**), CRISPR arrays (**c**), virulent factor and antimicrobial resistance gene (**d**), as well as protein transmembrane topology (**e**).
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/FuncDemo.png" width="700"></div>

+ **The PhageScope visualization of the sequence alignment.** PhageScope provides the linear (**a**) and circular (**b**) alignment visualizations. The color depth of the gray strips indicates the alignment coverage, whereas the color depth of the vertical blocks indicates the alignment identity.
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/AlignDemo.png" width="600"></div>

+ **The PhageScope visualization of the comparative tree.** The colors of the outermost circle denote the phage cluster, whereas the colors of the leaf node denote the phage subcluster.
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/TreeDemo.png" width="600"></div>

+ **The PhageScope visualization of the comparative tree with protein functional class annotated.** PhageScope also provides visualizations incorporating protein annotation and comparative tree.
<div align=center><img src="https://github.com/deepomicslab/PhageScope/blob/main/Tutorial/Vis_docs/Vis_figures/TreeheatmapDemo.png" width="500"></div>

